federation/scim/src/main/java/org/keycloak/federation/scim/core/ScimEndpointConfigurationStorageProviderFactory.java

```diff
@@ -0,0 +1,118 @@
+                .options(MediaType.APPLICATION_JSON, HttpHeader.SCIM_CONTENT_TYPE).defaultValue(HttpHeader.SCIM_CONTENT_TYPE)
+                .add().property().name(ScrimEndPointConfiguration.CONF_KEY_AUTH_MODE).type(ProviderConfigProperty.LIST_TYPE)
+                .label("Auth mode").helpText("Select the authorization mode").options("NONE", "BASIC_AUTH", "BEARER")
+                .defaultValue("NONE").add().property().name(ScrimEndPointConfiguration.CONF_KEY_AUTH_USER)
+                .type(ProviderConfigProperty.STRING_TYPE).label("Auth username").helpText("Required for basic authentication.")
+                .add().property().name(ScrimEndPointConfiguration.CONF_KEY_AUTH_PASSWORD).type(ProviderConfigProperty.PASSWORD)
+                .label("Auth password/token").helpText("Password or token required for basic or bearer authentication.").add()
+                .property().name(ScrimEndPointConfiguration.CONF_KEY_PROPAGATION_USER).type(ProviderConfigProperty.BOOLEAN_TYPE)
+                .label("Enable user propagation").helpText("Should operation on users be propagated to this provider?")
+                .defaultValue(BooleanUtils.TRUE).add().property().name(ScrimEndPointConfiguration.CONF_KEY_PROPAGATION_GROUP)
+                .type(ProviderConfigProperty.BOOLEAN_TYPE).label("Enable group propagation")
+                .helpText("Should operation on groups be propagated to this provider?").defaultValue(BooleanUtils.TRUE).add()
+                .property().name(ScrimEndPointConfiguration.CONF_KEY_SYNC_IMPORT).type(ProviderConfigProperty.BOOLEAN_TYPE)
+                .label("Enable import during sync").add().property()
+                .name(ScrimEndPointConfiguration.CONF_KEY_SYNC_IMPORT_ACTION).type(ProviderConfigProperty.LIST_TYPE)
+                .label("Import action").helpText("What to do when the user doesn't exists in Keycloak.")
+                .options("NOTHING", "CREATE_LOCAL", "DELETE_REMOTE").defaultValue("CREATE_LOCAL").add().property()
+                .name(ScrimEndPointConfiguration.CONF_KEY_SYNC_REFRESH).type(ProviderConfigProperty.BOOLEAN_TYPE)
+                .label("Enable refresh during sync").name(ScrimEndPointConfiguration.CONF_KEY_LOG_ALL_SCIM_REQUESTS)
+                .type(ProviderConfigProperty.BOOLEAN_TYPE).label("Log SCIM requests and responses")
+                .helpText("If true, all sent SCIM requests and responses will be logged").add().build();
+    }
+
+    @Override
+    public ScimEndpointConfigurationStorageProvider create(KeycloakSession session, ComponentModel model) {
+        return new ScimEndpointConfigurationStorageProvider();
+    }
+
+    /**
+     * Empty implementation : we used this {@link ScimEndpointConfigurationStorageProviderFactory} to generate Admin Console
+     * page.
+     */
+    public static final class ScimEndpointConfigurationStorageProvider implements UserStorageProvider {
```

Comment:
I had a chat with @stianst about this one and we go with a `UserStorageProvider` implementation instead of an event listener.

The reasons are that we should be able to deliver much more federation capabilities as well as limitations when using event listeners. For instance, you cannot delete users without a `UserRegistrationProvider` implementation. Also, event listeners are executed after removing resources which makes it hard to fetch data from these resources other than sharing data using an event detail.